README-dev.md

```diff
@@ -82,24 +82,31 @@ o1js uses these types to ensure that the constants used in the protocol are cons
 
 ## Development
 
-### Branch Compatibility
+### Branching Policy
 
-If you work on o1js, create a feature branch off of one of these base branches. It's encouraged to submit your work-in-progress as a draft PR to raise visibility! When working with submodules and various interconnected parts of the stack, ensure you are on the correct branches that are compatible with each other.
+| o1js base branches | Is default? |
+| ------------------ | ----------- |
+| main               | **Yes**     |
+| develop            | No          |
 
-#### How to Use the Branches
+When you start your work on o1js, please create the feature branch off of one of the above base branches.  
+It's encouraged to submit your work-in-progress as a draft PR to raise visibility!  
+When working with submodules and various interconnected parts of the stack, ensure you are on the correct branches that are compatible with each other.
 
-**Default to `main` as the base branch**.
+Other base branches (currently `develop` only) are used in specific scenarios where you want to adapt o1js to changes in the sibling repos on those other branches. Even then, consider whether it is feasible to land your changes to `main` and merge to `develop` afterwards. Only changes in `main` will ever be released, so anything in other branches has to be backported and reconciled with the `main` branch eventually.
```

Comment:
even though it might feel redundant because you already have it in the table -- please keep the sentence

> **Default to `main` as the base branch**.

in its original place here. it's much clearer as a full sentence than as a table entry "Is default? Yes", and provides more context to this paragraph where we start talking about "other" base branches (other than what?).

Reply:
Done. 

> other than what?

than listed above that paragraph :)